pkg/cloudprovider/cloudprovider.go

```diff
@@ -194,7 +198,11 @@ func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *v1alpha5.
 	if err != nil {
 		return false, client.IgnoreNotFound(fmt.Errorf("resolving node template, %w", err))
 	}
-	return c.isNodeTemplateDrifted(ctx, machine, provisioner, nodeTemplate)
+	drifted, err := c.isNodeTemplateDrifted(ctx, machine, provisioner, nodeTemplate)
+	if err != nil {
+		return false, cloudprovider.IgnoreMachineNotFoundError(err)
```

Comment:
I think we should keep it as is here. If the instance is missing and the node is orphaned, we should let the garbage collection handle this, rather than trying to get the Drift node controller handle it.

Reply:
Right, but my thought is that you should return errors from the interface as transparently as possible. If the instance is NotFound, that's an error that the client should be aware of and should be able to handle how they like. Think if we have some other package that is calling this package. Should we obfuscate that the instance isn't found and that's why we are returning false? IMO, I don't think we should.

Reply:
Yeah, that sounds good

Reply:
https://github.com/aws/karpenter-core/pull/373